### PR TITLE
add basic UI funtionalities

### PR DESCRIPTION
**Pull request by Mikelobi**
In this branch I added functionality to:

- [ ] Presents the user with a list of options to perform.
- [ ] Lets users choose an option.
- [ ] Asks user to input parameters for the option.
- [ ] List all books.
- [ ] List all people.
- [ ] Create a person (teacher or student).
- [ ] Create a book.
- [ ] Create a rental.
- [ ] List all rentals for a given person id.
- [ ] Gives users a way to quit the app.
- [ ] fix linter error

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+  end
+
+  def create_student
+    age = person_age_input
+    name = person_name_input
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase
+    puts 'Thank you for giving permission' if parent_permission
+    student = Student.new(age, name, @class)
+    @people << student
```
Comment (thecodechaser) [inline app.rb]: - [ ] When I try to create a student the application crashes, Please fix this issue. 

https://github.com/Mikelobi/oop-school-library/blob/e5da2a1a159d95ed6841dacca94515a9735f4c7a/student.rb#L6-L9

To fix this issue on the file student.rb, Replace the initializer line code with this `def initialize(classroom, age, name = 'unknown', parent_permission: true)`

- [ ] There's also another issue, The student initializer accepts `classroom, age, name` as parameters and you are passing `age, name, classroom`, You are swapping the parameters. Please fix this issue and pass the parameters in this sequence `Student.new(@class, age, name)`

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+  end
+end
+
+def create_teacher
+  age = person_age_input
+  name = person_name_input
+  print 'Area of specialization [e.g. Physics]: '
+  specialization = gets.chomp
+  teacher = Teacher.new(age, name, specialization)
+  @people << teacher
```
Comment (thecodechaser) [inline app.rb]: - [ ] When I try to create a teacher the application crashes, Please fix this issue. 

https://github.com/Mikelobi/oop-school-library/blob/e5da2a1a159d95ed6841dacca94515a9735f4c7a/teacher.rb#L4-L7

To fix this issue on the file teacher.rb, Replace the initializer line code with this `def initialize(specialization, age, name = 'unknown', parent_permission: true)`

- [ ] There's also another issue, The teacher initializer accepts `specialization, age, name` as parameters and you are passing `age, name, specialization`, You are swapping the parameters. Please fix this issue and pass the parameters in this sequence `Teacher.new(specialization, age, name)`.

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+def list_rentals
+  if @rentals.empty?
+    puts 'Please add a rental first.'
+    create_rental
+  else
+    print 'Enter ID of person: '
+    person_id = gets.chomp.to_i
+    sleep 0.75
+    rental_list = @rentals.select { |rental| rental.person.id == person_id }
```
Comment (thecodechaser) [inline app.rb:192]: - [ ] Please display the data(ID) of persons after the code `print 'Enter ID of person: '` so users can enter to person ID to list the rentals for that person. Right now there's no data displayed for persons so users will not be able to know the user's ID to get rentals, and we are generating the IDs randomly.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @Mikelobi👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KayLemba — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hello  @Mikelobi        , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

HIGHLIGHTS:

- Your project is professional 
- Your repo is professional 
- Well structured files 


Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 

well-1



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**